.github/workflows/create_release.yml

```diff
@@ -0,0 +1,178 @@
+            
+    permissions:
+      contents: write  # IMPORTANT: mandatory for making GitHub Releases # TODO: check what is needed here?
+      id-token: write  # IMPORTANT: mandatory for trusted publishing (which means without api-token or password)
+
+    steps:
+
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: wheels* # TODO change back to python-wheels?
+          path: dist
+          merge-multiple: true
+
+      - name: Publish distribution to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          repository-url: https://test.pypi.org/legacy/
+          verbose: true     
+          print-hash: true
+
+
+
+
+  prepare-release:    
+    name: Release-Prep (p.ex. sigstore, pypi)
+    runs-on: ubuntu-latest
+    needs: [call-workflow-mac, call-workflow-ubuntu_x86, call-workflow-ubuntu_aarch64, call-workflow-win]
+    if: (needs.call-workflow-mac.result == 'success') && (needs.call-workflow-ubuntu_x86.result == 'success') && (needs.call-workflow-ubuntu_aarch64.result == 'success')  && (needs.call-workflow-win.result == 'success') 
+
+    permissions:
+      contents: write  # IMPORTANT: mandatory for making GitHub Releases
+      id-token: write  # IMPORTANT: mandatory for sigstore
+
+    steps:
+    
+      - uses: actions/download-artifact@v4
+        with:
+          pattern: wheels* # TODO change back to python-wheels?
+          path: dist
+          merge-multiple: true
+
+      - name: Sign the dists with Sigstore #/home/runner/work/onnx/onnx/dist/*.tar.gz
+        uses: sigstore/gh-action-sigstore-python@v3.0.0
+        with:
+          inputs: >-            
+            /home/runner/work/onnx/onnx/dist/*.whl
+  
+      #- name: Rename files # to match new file extension https://github.com/sigstore/sigstore-python/blob/main/CHANGELOG.md#changed
+      #  run: |
+      #    sudo apt install mmv 
+      #    mmv "/home/runner/work/onnx/onnx/dist/*.sigstore" /home/runner/work/onnx/onnx/dist/#1.sigstore.json      
+      
+      - uses: actions/upload-artifact@0b2256b8c012f0828dc542b3febcab082c67f72b
+        with:
+          name: sigstore-files
+          path: |
+              /home/runner/work/onnx/onnx/dist/*.sigstore.json
+
+                      
+      # TODO
+      # at this point, we have the wheels and could check if they are usable by offline testing...continue-on-error: 
+      # For more information about environments and required approvals, see "Using environments for deployment." F
+      # We can use a separate requirement for deploy
+
+  release: 
+    # TODO Add a deployment step for reviewing https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments
+    name: Release (Publish to pypi and add files to github release)
+    runs-on: ubuntu-latest
+    needs: [prepare-release]
+    if: always() && (needs.prepare-release.result == 'success') && startsWith(github.base_ref, 'rel-') # TODO only if a release is requested
```

Comment:
Is it here that we should check that the trigger is scheduled?

Reply:
What exactly do you mean by that? 

I think one should take another look at all my steps to see if the checks were chosen correctly.

I currently also assume that the Github release is generated via the GUI and I only add the files in this step accordingly. In this release step, you would now also have to define a Github deployment environment, which you link to Pypi, so that the reviewers first have to approve the step before the Whl is actually pushed. I think you would have 30 days for the review, you can download the wheels beforehand for local testing. Maybe the handling of the RC candidates is currently missing? Do we also want them on pypi? or would they be better off on testpypi?